example/models/food/v1/food_test.go

```diff
@@ -0,0 +1,138 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/ductone/protoc-gen-pgdb/internal/pgtest"
+	pgdb_v1 "github.com/ductone/protoc-gen-pgdb/pgdb/v1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSchemaFoodPasta(t *testing.T) {
+	ctx := context.Background()
+	pg, err := pgtest.Start()
+	require.NoError(t, err)
+	defer pg.Stop()
+
+	_, err = pg.DB.Exec(ctx, "CREATE EXTENSION IF NOT EXISTS btree_gin")
+	require.NoError(t, err)
+
+	smsg := &Pasta{}
+	schema, err := pgdb_v1.CreateSchema(smsg)
+	require.NoError(t, err)
+	for _, line := range schema {
+		fmt.Printf("%s \n", line)
+		_, err := pg.DB.Exec(ctx, line)
+		require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to execute sql: '\n%s\n'", line)
+		// os.Stderr.WriteString(line)
+		// os.Stderr.WriteString("\n------\n")
+	}
+	ct := schema[0]
+	require.Contains(t, ct, "CREATE TABLE")
+	require.Equal(t, 2,
+		strings.Count(ct, "$pksk"),
+		"Create table should contain only one pksk field + index: %s", ct,
+	)
+
+	require.Contains(t, ct, "PARTITION BY LIST")
+
+	require.Equal(t, 1,
+		strings.Count(ct, "fts_data"),
+		"Create table should contain only one fts_data field: %s", ct,
+	)
+	_, err = pg.DB.Exec(ctx, "DROP TABLE "+smsg.DBReflect().Descriptor().TableName())
+	require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to drop")
+
+	schema, err = pgdb_v1.Migrations(ctx, pg.DB, smsg)
+	require.NoError(t, err)
+	for _, line := range schema {
+		_, err := pg.DB.Exec(ctx, line)
+		require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to execute sql: '\n%s\n'", line)
+		// os.Stderr.WriteString(line)
+		// os.Stderr.WriteString("\n------\n")
+	}
+	ct = schema[0]
+	require.Contains(t, ct, "CREATE TABLE")
+
+	fakeTenantIds := []string{"t1", "t2", "t3"}
+	protoTableName := smsg.DBReflect().Descriptor().TableName()
+
+	// TODO(scott) makes helper functions and move to pgdb_v1
```

Comment:
yeah, can we explore what the `pgdb_v1.Migrations` equivalent function would be for table attachment?

could it be something that has a tenant iterator (so we can do it incrementally, or for only a single tenant at a time)?

```
err = pgdb_v1.TenantPartitionsUpdate(ctx, pg.DB, smsg,
     func(ctx context.Context) (string, error) {
          // iterator function, returns a TenantID or error.  On Empty String return, iteration is complete.
         return "12345", nil
     },
    func(ctx context.Context, tenantID string, schema string) error {
       // schema update function, called for each CREATE/ATTACHMENT or any future migraiton pattern needed on a per-tenant basis of the partition.
         return pg.DB.Exec(ctx, schema)
    }
 )
```

Reply:
For context of what I've found

Altering a column (add / drop) to the master table will apply to all sub tables atuomatically

Indexing I believe we need to do per table.

Reply:
https://github.com/ductone/protoc-gen-pgdb/pull/63/commits/2e2550bfa23a80c7e07d548ec041b01323208fb7 done here